### PR TITLE
Add a helper class that reads metrics every X seconds.

### PR DESCRIPTION
**Pull request by jkwatson**
This is a final bit of clean up from @bogdandrutu's work in #944

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  /**
+   * Stops the scheduled task and calls export one more time.
+   *
+   * @since 0.3.0
+   */
+  public void shutdown() {
+    scheduler.shutdown();
+    exporter.run();
+  }
+
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java]: We said that we need to wait for the in-progress task if any.

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java]: I don't think we ever came to a conclusion on how we wanted to behave, but I'm happy to have this method block until everything is done, if that's what is desired.

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java]: How long should we wait before bailing out, do you think?

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java]: I made it wait 5 seconds. Happy to tweak it if there's a better number.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    scheduler.shutdown();
+    try {
+      scheduler.awaitTermination(5, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      // force a shutdown if the export hasn't finished.
+      scheduler.shutdownNow();
+      // reset the interrupted status
+      Thread.currentThread().interrupt();
+    } finally {
+      exporter.run();
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java]: If we cannot stop the previous export we should not start a new one :)

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java]: yeah, good call. Fixed.

**Pull request merged**